Sources/NetworkService/NetworkError/NetworkError.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// Contains network errors that may occur, while the application is running.
+public enum NetworkError: Error, CaseIterable {
+    /// Indicates that the Internet connection has been lost.
+    case lostConnection
+    
+    /// Indicates that the server cannot handle the request (because it is overloaded or down for maintenance).
+    case serverUnavailable
+    
+    /// Indicates that the server cannot process the request due to a client error
+    /// (e.g., malformed request syntax, size too large, invalid request message framing, or deceptive request routing).
+    case badRequest
+    
+    /// Indicates that the request to connect and log on to an FTP server could not be completed
+    /// because the supplied password is incorrect.
+    case passwordError
```

Comment:
In my opinion loginError and passwordError represent errors related to the authentication process.
```suggestion
        /// Indicates an authentication error occurred, either due to a login failure or an incorrect password.
    case authenticationError
```

Reply:
Thanks for the suggestion. Fixed it.